### DISCUSSION TITLE
Question about Scheduler and repeat

### DISCUSSION THREAD

**Discussion by Zimtente** [taskforcesh/bullmq#2963] [General]
Hey there,

we are using bullmq for quite some time now. Recently we started having the issue that in some queues we had multiple delayed jobs. this queue just has 1 scheduler creating 1 job every 10 seconds. The job runs for around 100-250ms. How can it happen that there are multiple delayed jobs in our queue? Even considering that the job >may< run for longer than 10 seconds (as there is only 1 worker on this queue) it just seems like once there is a second job in "delayed" that we have 2 tasks being executed and recreated by the scheduler instead of just 1. 

We double checked only having 1 scheduler. Being on the latest version of bullmq etc. but did not have any success fixing it. Restarting our cluster seems to have high probability to have this "issue" shown up. 

We are using upsertJobScheduler to setup our queues with unique job name and we double checked that there is no second scheduler. 

So our thinking is... that we fundamentally use repeating tasks in a wrong way. Anyone can help with this?

Thank you!

**Comment by manast** [comment]
Do you have any code that reproduces the issue? we had a similar issue some time ago but has been fixed. Are you sure you are using the latest version when this happens?

**Reply by Zimtente** [reply]
thx for the reply. yes we also tested it on 5.34.0

I can print the code we are using. but generally just creating or finding existing a queue and then upserting a scheduler with repeating jobOptions. But maybe first showing some screenshots from TaskForce.

![image](https://github.com/user-attachments/assets/3ef0f3f9-cd39-4ecf-b27d-2e56d50b43d8)

![image](https://github.com/user-attachments/assets/7d934ee7-783e-4bc7-afdc-afb7cfa92aa7)

![image](https://github.com/user-attachments/assets/2d992c72-2e2c-403a-96f4-67768f97bf31)

I think we may not understand how the scheduler works. I thought it works like this: there is 2 delayed jobs (I guess could happen if we restart) the worker should just complete these 2. After that the scheduler should create 1 new one after 60 seconds? 

But this does not seem to be the case. Worker completes 2 jobs and then scheduler creates 2 more. Once 2 jobs in delayed it always will be 2. and so on.

**Reply by Zimtente** [reply]
could it be that we somehow mixed the old way of repeating and the new way with the scheduler?

**Reply by manast** [reply]
ok. This issue will not be able to happen again as we are removing the old repeatable jobs mechanism for v6

**Reply by Zimtente** [reply]
yes basically mixed it by passing default job options with repeating into the scheduler upsert. I did verify it by now. sorry for late reply wasnt 100% sure first if that was the issue.

![image](https://github.com/user-attachments/assets/4910931d-5f7a-456a-9c5a-65cb27196ccd)

**Reply by manast** [reply]
Ok, does it mean that the issue you had in the beginning is resolved then?

**Reply by Zimtente** [reply]
yes
